### ISSUE TITLE
Rewrite & cleanup

### ISSUE THREAD

**Issue by zanieb** [mikeza/jrivermcws#2]
This is a pretty hacked together project right now -- should be able to rewrite it in a weekend and make it a lot better.

Planning on sometime before the end of August.

**Comment by kenomaerz** [comment]
I started an implementation for python as well last year. I just released version 1.0.0 here: https://github.com/kenomaerz/pymcws

It already has a number of nice features and is in heavy use in my ecosystem, but it doesn't have CLI features. If you are interested in teaming up, I would love to hear your thoughts.

**Comment by zanieb** [comment]
> I started an implementation for python as well last year. I just released version 1.0.0 here: https://github.com/kenomaerz/pymcws
> 
> It already has a number of nice features and is in heavy use in my ecosystem, but it doesn't have CLI features. If you are interested in teaming up, I would love to hear your thoughts.

Sweet! I like your project. It looks like you've manually wrapped a lot of the API--which makes sense because it's hard to work with but I'm not sure I'd have time to help maintain it. As far as a CLI, I bet you could generate commands for your API at `ApiMediaServer` with `click` and some magic.

I'm pretty busy with my job and such but hopefully have some time to address this eventually still... my wish list includes
- [ ] Click CLI with generated API commands.
- [ ] Autogenerated API routes for each version of the media center
- [ ] Better automation for API result parsing
- [ ] Full rewrite of existing code with tests / CI

I mostly use this for displaying the current track, rating the current track with a hotkey, and scraping genre metadata from online and updating it in jriver -- making genre scraping useful to others is a stretch goal as well. I'm not sure if the number of people using it merits the time it'd take to make it robust.

**Comment by kenomaerz** [comment]
Yes, making the API "pythonian" requires lots of manual wrapping. My main goal was to make working with fields equally powerful as in MC while making it pain-free and type-safe in python. I definitely get the time constraints, it's the same for me here. I mostly saw that you intended to rewrite your code and wondered whether basing the api calls on pymcws was an option - similar to what you suggested with click. That would take care of API result parsing and all type conversions, since that is already in pymcws, and it would definitely improve the package with a functionality I cannot provide right now. I could add the routes you need as well too, since I can basically generate them from the docs now. 

I took supporting all API versions off the scope for me personally because the changes are so complex in how they interact just for the last few versions that I figured I could take a stab at it when someone first addresses the need.

Either way, maybe sleep on it, but no pressure! Much success with your project :)
